portal-api/src/v1/users/routes.js

```diff
@@ -209,20 +225,7 @@ module.exports.login = async (req, res, next) => {
   }
 
   if (FEATURE_FLAGS.MAGIC_LINK) {
-    findByUsername(username, async (error, user) => {
-      if (error) {
-        next(error);
-      } else if (user) {
-        const { status } = await sendSignInLinkEmail(user.email, user.firstname, 'placeholderSignInLink');
-        if (status === 201) {
-          // TODO DTFS2-6680: Add success logic here.
-        } else {
-          // TODO DTFS2-6680: Add failure logic here.
-        }
-      } else {
-        next(usernameOrPasswordIncorrect);
-      }
-    });
+    findByUsername(username, sendSignInLinkEmailAndHandleErrors(next));
```

Comment:
Do we not need to `await` `findByUsername`?

Reply:
I spoke to @oscar-richardson-softwire on a call about this and the answer is yes but we're not going to do so in this PR because this code is about to change anyway